app/src/main/java/com/tari/android/wallet/ui/activity/CreateWalletActivity.kt

```diff
@@ -0,0 +1,208 @@
+    @BindView(R.id.crypto_wallet_main_txt2)
+    lateinit var cryptoWalletMainText2: TextView
+    @BindView(R.id.crypto_wallet_des_txt)
+    lateinit var cryptoWalletDesText: TextView
+    @BindView(R.id.create_wallet_btn)
+    lateinit var createWalletButton: TextView
+    @BindView(R.id.create_wallet_btn_layout)
+    lateinit var walletBtnLayout: FrameLayout
+    @BindView(R.id.viewContainer)
+    lateinit var viewContainer: RelativeLayout
+    @BindView(R.id.txt_testnet)
+    lateinit var testnetTextView: TextView
+    @BindView(R.id.main_img_small_gem)
+    lateinit var smallGemImageView: ImageView
+    @BindView(R.id.progress)
+    lateinit var progressBar: ProgressBar
+    @BindView(R.id.videoView_loop1)
+    lateinit var videoView1: VideoView
+    @BindView(R.id.videoView_loop2)
+    lateinit var videoView2: VideoView
+    @BindView(R.id.video_container)
+    lateinit var videoContainer: FrameLayout
+
+    @BindDimen(R.dimen.splash_screen_title_bottom_margin)
+    @JvmField
+    var mainContentShiftMargin = 0
+
+    override val contentViewId = R.layout.activity_create_wallet
+
+    private var videoLoopCount = 0
+    private val videoLoopHandler = Handler()
+    private val uiHandler = Handler()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
+        init()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        videoView1.setVideoURI(getResourceUri(R.raw.splash_video_loop_1))
+        videoView2.setVideoURI(getResourceUri(R.raw.splash_video_loop_2))
+        videoView1.setOnPreparedListener {
+            // setVideoLoopNotifier(videoView1.duration)
+        }
+        videoView1.start()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        videoLoopHandler.removeCallbacksAndMessages(null)
+        uiHandler.removeCallbacksAndMessages(null)
+    }
+
+    private val videoLoopRunnable: Runnable = Runnable {
+        videoLoopCount = if (++videoLoopCount < 2) videoLoopCount else 0
+
+        if (videoLoopCount == 0) {
+            videoView2.pause()
+            videoView1.seekTo(2000)
+            videoView1.start()
+        } else {
+            videoView1.pause()
+            videoView2.seekTo(2000)
+            videoView2.start()
+        }
+
+        setVideoLoopNotifier(videoView1.duration)
+    }
+
+    private fun setVideoLoopNotifier(duration: Int) {
+        if (duration == -1) return
+        videoLoopHandler.removeCallbacks(videoLoopRunnable)
+        videoLoopHandler.postDelayed(videoLoopRunnable, (duration - 3000).toLong())
+    }
+
+    private fun init() {
+        rootView.post { startViewAnimation() }
+        setVideoDimension()
+    }
+
+    private fun setVideoDimension() {
+        rootView.viewTreeObserver.addOnGlobalLayoutListener(
+            object : ViewTreeObserver.OnGlobalLayoutListener {
+                override fun onGlobalLayout() {
```

Comment:
I think you should remove the the global layout listener at the end of this function once you're done with it. Also here again: careful with retain cycles.

Reply:
Yes, we are removing it at the beginning of the function `rootView.viewTreeObserver.removeOnGlobalLayoutListener(this)`.

Reply:
Sorry I failed to see it.